libs/langchain/langchain/indexes/base.py

```diff
@@ -1,172 +1,5 @@
-from __future__ import annotations
+from langchain_community.indexes.base import RecordManager
 
-import uuid
-from abc import ABC, abstractmethod
-from typing import List, Optional, Sequence
-
-NAMESPACE_UUID = uuid.UUID(int=1984)
-
-
-class RecordManager(ABC):
```

Comment:
This should stay in langchain or end up in langchain core since partner packages will need to inherit for their own   implementations.

Generally abstract interfaces should be in either langchain core or langchain, but not in langchain community

Reply:
@eyurtsev  Yeah, exactly, that was implemented in the #17832 but @baskaryan [opposed](https://github.com/langchain-ai/langchain/pull/17832#pullrequestreview-1894644198) it. So...
looks like you and @baskaryan should decide what to do or not to do. I'm pausing now with this PR.